pkg/runtime/reconciler.go

```diff
@@ -187,12 +187,39 @@ func (r *reconciler) sync(
 			"arn", latest.Identifiers().ARN(),
 			"is_adopted", isAdopted,
 		)
+		// Before we update the backend AWS service resources, let's first update
+		// the latest status of CR which was retrieved by ReadOne call.
+		// Else, latest read status is lost in-case Update call fails with error.
+		err = r.updateCRStatus(ctx, desired, latest)
+		if err != nil {
+			return err
+		}
 		latest, err = rm.Update(ctx, desired, latest, diffReporter)
 		if err != nil {
 			return err
 		}
 		r.log.V(0).Info("reconciler.sync updated resource")
 	}
+	err = r.updateCRStatus(ctx, desired, latest)
+	if err != nil {
+		return err
+	}
+	for _, condition := range latest.Conditions() {
+		if condition.Type == ackv1alpha1.ConditionTypeResourceSynced &&
+			condition.Status != corev1.ConditionTrue {
+			return requeue.NeededAfter(
+				errors.New("sync again"), requeue.DefaultRequeueAfterDuration)
```

Comment:
Instead of using `errors.New()` here, please create a new error variable in the `pkg/errors` package:

https://github.com/aws/aws-controllers-k8s/blob/main/pkg/errors/error.go

How about we call this error `TemporaryOutOfSync` or something like that, to indicate the error isn't really an error but more of a marker that the status check will be performed after some wait time?